subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/attributes/DefaultAttributesSchemaTest.groovy

```diff
@@ -78,8 +78,8 @@ class DefaultAttributesSchemaTest extends Specification {
         schema.attribute(attribute)
 
         expect:
-        !schema.mergeWith(EmptySchema.INSTANCE).matchValue(attribute, "a", "b")
-        schema.mergeWith(EmptySchema.INSTANCE).matchValue(attribute, "a", "a")
+        !new DefaultAttributeSelectionSchema(schema, EmptySchema.INSTANCE).matchValue(attribute, "a", "b")
```

Comment:
At what point are we no longer testing the DefaultAttributesSchema and we're really testing DefaultAttributeSelectionSchema here?

Would it make more sense to make schema a DefaultAttributeSelectionSchema at the top?

Reply:
Most of this logic does test `DefaultAttributeSelectionSchema`. TBH most of these tests should be moved out of this class, but thats for a later PR. I want to limit changes here.